app/controllers/cinema_halls_controller.rb

```diff
@@ -21,13 +21,14 @@ def create
   end
 
   def show
-    @cinema_hall = CinemaHall.find(params[:id])
+    @cinema_hall = CinemaHalls::Representers::Single.new(id: params[:id]).call
```

Comment:
You should pass here a specific object, not an ID. The responsibility of `Representer` is to `represent` some object which is already pushed to this class.